src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -1433,3 +1513,72 @@ def merge_metadata(a_metadata, b_metadata, *, source, update=(), min_=(), max_=(
         result_directory=r_a.result_directory,
         app_version_info=app_version_info,
     )
+
+
+@contextmanager
+def open_global_sensitivity_analysis_result_file(
+    results_directory: Path,
+) -> Optional[Iterator[h5py.File]]:
+    """
+    Open a global sensitivity analysis result file.
+    """
+    filename = results_directory / "uq_result"
+    ignored_file = results_directory / "uq_result.creating"
+    # Avoid to read result files with incomplete metadata.
+    if not ignored_file.is_file():
+        file = _open_result_file(filename)
+        yield file
+        file.close()
+    else:
+        yield
+
+
+def read_global_sensitivity_analysis_meta_data(
+    result_directory: Path,
+) -> Optional[GlobalSensitivityAnalysisMetaData]:
+    """
+    Read the global sensitivity analysis metadata persisted in a result file.
+    """
+
+    return GlobalSensitivityAnalysisMetaData.get_meta_data_from_dir(
+        result_directory=result_directory
+    )
+
+
+def read_read_global_sensitivity_analysis_time_set(
+    result_directory: Path,
+) -> Optional[numpy.array]:
+    """
+    Get the time set for sensitivity analysis results.
+    """
+    with open_global_sensitivity_analysis_result_file(
+        results_directory=result_directory
+    ) as result_file:
+        if not result_file:
+            return
+        return result_file[GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME]["time_set"][:]
+
+
+def read_global_sensitivity_coefficients(
+    property_id: str, parametric_var_id: str, trend_id: str, results_directory: Path
+) -> Optional[Dict[str, numpy.array]]:
+    """
+    Read the global sensitivity analysis coefficients results.
+    """
+    uq_meta_data = read_global_sensitivity_analysis_meta_data(results_directory)
+    with open_global_sensitivity_analysis_result_file(results_directory) as result_file:
+        if not result_file:
+            return
+        global_sensitivity_analysis_coefficients = result_file[
+            GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME
+        ]["global_sensitivity_analysis"][:]
+
+    coefficients_key = f"{property_id}::{parametric_var_id}@{trend_id}"
+    meta_information = uq_meta_data.global_sensitivity_analysis_meta_data[
+        coefficients_key
+    ]
+    return {
+        coefficients_key: global_sensitivity_analysis_coefficients[
+            meta_information.qoi_index, meta_information.qoi_data_index
+        ]
+    }
```

Comment:
Why returning a dict if it has only one key? Maybe a tuple with the coefficient and the results would be more intuitive for users of this function.
```suggestion
    return (
        coefficients_key, 
        global_sensitivity_analysis_coefficients[
            meta_information.qoi_index, meta_information.qoi_data_index
        ]
    )
```